Website: Add upgrade guide and example (#583)
metallb/metallb: website/content/installation/_index.md

generatorOptions:
 disableNameSuffixHash: true
```

## Upgrade Current Version

> To upgrade current version of metallb to new release, always look at [the release notes](https://metallb.universe.tf/release-notes/) to see the changes and action required items. Pay special attention when upgrading major/minor releases.

Reviewer: Not sure if it's me not being fully awake yet, or find the beginning of the sentence confusing. Would something like this be more clear to you?

```suggestion
When upgrading MetalLB, it is recommended to always look at [the release notes](https://metallb.universe.tf/release-notes/) for the changes and action required items. Pay special attention when upgrading major/minor releases.
```

Reviewer 2: "action required items" should be either "action-required items" or "required action items".  Also "it is recommended to" can be deleted.

Other than that, LGTM.

> After that we can install new metallb version by downloading the new manifests and `kubectl apply -f` them.

Reviewer: What if we change `we` to `you`, mention kubectl and kustomize (as both install options are listed in the docs now) and also mention that if we happen to remove a resource (while not frequent at all, it might happen) the user needs to take care of it manually.
```suggestion
After that, usually you can install the new metallb version by just downloading the new manifests and either use `kubectl` or kustomize to apply them. In rare cases, you will need to manually remove a kubernetes resource if it is not present in the new manifests (e.g. if a resource `kind: ClusterRole` is removed because the permission is not needed anymore).
```

Reviewer 2: LGTM

generatorOptions:
 disableNameSuffixHash: true
```

> ## Upgrade Current Version

Reviewer: Just `Upgrade`?

Author: Yes, at the moment

Reviewer: Sorry I wasn't clear, I meant this:
```suggestion
 ## Upgrade
```

Does it make sense? (i.e. just call the section "upgrade")

> To upgrade current version of metallb to new release, always look at [the release notes](https://metallb.universe.tf/release-notes/) to see the changes and action required.

Reviewer: ```suggestion
To upgrade current version of metallb to new release, always look at [the release notes](https://metallb.universe.tf/release-notes/) to see the changes and action required items. Pay special attention when upgrading major/minor releases.
```

> After that we can install new metallb version by downloading the new manifests and `kubectl apply -f` them. For example we upgrade from metallb v0.8.1 to v0.9.3. By reading [the release notes](https://metallb.universe.tf/release-notes/) we know that newer metallb use new algorithm to detect dead nodes. That algorithm need `secretkey` that we need to create. Here the steps to upgrade metallb v0.8.1 to v0.9.3:

Reviewer: What if we just mention the generalities, without an example?

```suggestion
After that we can install new metallb version by downloading the new manifests and `kubectl apply -f` them.
 
Take into account the known limitations for [layer2](https://metallb.universe.tf/concepts/layer2/#limitations) and [bgp](https://metallb.universe.tf/concepts/bgp/#limitations) when doing so.
```

Author: Ah, I agree. That is good idea to make it more general.

Reviewer: Thanks!

```shell
# Create secret key
kubectl create secret generic -n metallb-system memberlist --from-literal=secretkey="$(openssl rand -base64 128)"
# apply new metallb manifest
kubectl apply -f https://raw.githubusercontent.com/google/metallb/main/manifests/metallb.yaml
```

then we can watch the metallb pod deletion and creation and `metallb-system` namespace:

```shell
watch kubectl -n metallb-system get po
```

> Upgrade is finish after all metallb pod in running status.

Reviewer: And remove all of this?

Author: I already remove all of that.

Reviewer: Thanks!
